prov/usnic/src/usdf_pep.c

```diff
@@ -658,6 +726,9 @@ usdf_pep_open(struct fid_fabric *fabric, struct fi_info *info,
 		break;
 	case FI_SOCKADDR_IN:
 		break;
+	case FI_ADDR_STR:
+		addr_format_str = true;
+		goto skip_size_check;
```

Comment:
Actually, instead of just skipping the size check, how about we move the size check inside the switch for the `FI_SOCKADDR` and `FI_SOCKADDR_IN` test. I'm not sure why we're only testing for `AF_INET` in `FI_SOCKADDR`. I think we should be doing it for both. 

So I'm suggesting: 

```c
switch (info->addr_format) {
case FI_SOCKADDR:
case FI_SOCKADDR_IN:
    /* check AF_INET */
    /* check size */
case FI_ADDR_STR:
    /* set str flag */
}
```